### ISSUE TITLE
Enhancement: Ability to configure more than one Azure AI Search Index

### ISSUE THREAD

**Issue by dareima** [danny-avila/LibreChat#3862]
### What features would you like to see added?

Ability to use more than one Azure AI Search Index in LibreChat, e. g. with modelSpecs.

### More details

An Azure AI Search indexes is usually built across a certain collection of data (documents).
Therefore it is common practise to have more than one index which can then be connected to an LLM as a search resource.

Currently, LibreChat supports only one specific Azure AI Search Index through the `.env` file.

Maybe it would be possible to provide multivalues for these params, e. g.:

AZURE_AI_SEARCH_SERVICE_ENDPOINT=https://ai-search-service1.search.windows.net
AZURE_AI_SEARCH_INDEX_NAME=ai-index-1,ai-index-2,ai-index-3
AZURE_AI_SEARCH_API_KEY=abcdefghijklmnopqrstuvwxyz

Ideally even more than one Search Service Endpoint could be configured but more than one index would already be a huge benefit. The ability to configure more than one Search Service would imply that the index names have to be assigned to their service and there would also be more than one Search API key.

Multiple indexes could then be combined with different modelSpecs as with the rework for plugins discussed in [3607](https://github.com/danny-avila/LibreChat/issues/3607) or some other way to select the desired index.

### Which components are impacted by your request?

Endpoints

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by RAPHCVR** [comment]
Nice Idea too

**Comment by mansiibm** [comment]
Yes, this will elevate the tool- Azure AI Search index.  @danny-avila are there any plans for this in the future releases.

**Comment by rubentalstra** [comment]
https://github.com/danny-avila/LibreChat/commit/edcac7669bfeb6fba8399fc748c52ee1a5ef00e9

**Comment by dareima** [comment]
> [edcac76](https://github.com/danny-avila/LibreChat/commit/edcac7669bfeb6fba8399fc748c52ee1a5ef00e9)

Fantastic, thanks so much for working on this @rubentalstra.

**Comment by nitroxeno** [comment]
This looks very promising, thanks for this @rubentalstra. Any way we could select a specific index? Ideally, in the modelSpecs preset?

**Comment by rubentalstra** [comment]
> This looks very promising, thanks for this @rubentalstra. Any way we could select a specific index? Ideally, in the modelSpecs preset? 

@nitroxeno you need a agent setup for this. But you can use an agent in the model specs so it would work.

**Comment by danny-avila** [comment]
Correcting the record here, because this thread reads as though the feature shipped: it did not.

Commit `edcac76` that @rubentalstra linked lived on the `refactor/Azure-AI-Search` branch behind PR #6156, which targeted `dev` and was **closed without merging** on 2025-07-21. Current main still has, in `api/app/clients/tools/structured/AzureAISearch.js`:

```js
this.indexName = this._initializeField(
  fields.AZURE_AI_SEARCH_INDEX_NAME,
  'AZURE_AI_SEARCH_INDEX_NAME',
);
```

a single `SearchClient` built from it, and a singular `AZURE_AI_SEARCH_INDEX_NAME=` in `.env.example`. So @dareima's original request is fully open.

@grmonteiroh, that is the update: no progress in main. @nitroxeno, on selecting a specific index, note that #6156 would not have given you that anyway. Its approach queried every configured index concurrently and flattened the results, which is a different feature from choosing an index per modelSpec or agent. Whoever revisits this should treat "accept multiple indexes" and "select which index a given spec uses" as two separate decisions.

Classification: enhancement, severity low, priority medium-low (3 upvotes plus three distinct askers, but requires an opt-in Azure setup).
